table/tables/tables_test.go

```diff
@@ -392,3 +392,94 @@ func (ts *testSuite) TestTableFromMeta(c *C) {
+	hiddenCols := tc.HiddenCols()
+	c.Assert(table.FindCol(hiddenCols, "a"), IsNil)
+	c.Assert(table.FindCol(hiddenCols, "b"), NotNil)
+	c.Assert(table.FindCol(hiddenCols, "c"), IsNil)
+	c.Assert(table.FindCol(hiddenCols, "d"), NotNil)
+	c.Assert(table.FindCol(hiddenCols, "e"), IsNil)
+
+	// Can't select with b and d
+	tk.MustQuery("select * from hidden;").Check(testkit.Rows("1 3 5"))
+	_, err = tk.Exec("select b from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'field list'")
+	_, err = tk.Exec("select d from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field list'")
+	err = tk.QueryToErr("select a, c, e from hidden;")
+	c.Assert(err, IsNil)
+	_, err = tk.Exec("select d from hidden;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field list'")
+	_, err = tk.Exec("select * from hidden where b>1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'where clause'")
+	_, err = tk.Exec("select * from hidden order by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'order clause'")
+	_, err = tk.Exec("select * from hidden group by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'group statement'")
+
+	// Can't update with b and d
+	_, err = tk.Exec("update hidden set d=1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'field_list'")
+	_, err = tk.Exec("update hidden set a=1 where b=1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'where clause'")
+	_, err = tk.Exec("update hidden set a=1 where c=2 order by b;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'order clause'")
+
+	// Can't delete with b and d
+	_, err = tk.Exec("delete from hidden where b=1;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'b' in 'where clause'")
+	_, err = tk.Exec("delete from hidden order by d;")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'd' in 'order clause'")
+
+	// Can't drop column b and d
+	_, err = tk.Exec("ALTER TABLE hidden DROP COLUMN b")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:1091]column b doesn't exist")
+	_, err = tk.Exec("ALTER TABLE hidden DROP COLUMN d")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:1091]column d doesn't exist")
+
+	// Test show create table
+	tk.MustQuery("show create table hidden;").Check(testkit.Rows(
+		"hidden CREATE TABLE `hidden` (\n" +
+			"  `a` int(11) NOT NULL,\n" +
+			"  `c` int(11) DEFAULT NULL,\n" +
+			"  `e` int(11) DEFAULT NULL,\n" +
+			"  PRIMARY KEY (`a`)\n" +
+			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
```

Comment:
Also, adding test for `select ... from inforschema`, check the visibility of hidden columns.

Reply:
`inforschema` is a view，this PR only handle TableCommon. Select with `inforschema` will be handled in next PR

Reply:
But now you deal with `infoschema` in this PR, could you add some test for it?

Reply:
It had been solved by #13954